src/pages/index.js

```diff
@@ -83,87 +82,109 @@ function handleDeleteIcon(cardId, card) {
     api
       .deleteCard(cardId)
       .then(() => {
-        deleteCardPopup.close();
         card.remove();
       })
       .catch((err) => {
         console.error(err);
+      })
+      .finally(() => {
+        deleteCardPopup.close();
       });
   });
 }
 
 function handleProfileImgFormSubmit(data) {
   api
     .updateProfileImg(data.link)
+    .then(profileImagePopup.editSubmitBtn())
     .then(() => {
       userInfo.setUserProfileImage(data.link);
-      profileImagePopup.close();
     })
-    .catch((error) => {
-      console.error("Error fetching data:", error);
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileImagePopup.resetSubmitBtn();
+      profileImagePopup.close();
     });
 }
 
 function handleCardClick(name, link) {
   previewImagePopup.open(name, link);
 }
-function addLike(cardId, likeCounter) {
+function addLike(cardId, likeCounter, likeBtn) {
   api
     .addLike(cardId)
     .then((res) => {
       likeCounter.textContent = res.likes.length;
+      console.log(likeBtn);
     })
-    .catch((error) => {
-      console.error("Error fetching data:", error);
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      likeBtn.classList.add("card__like-button_active");
```

Comment:
`      likeBtn.classList.add("card__like-button_active");
`

You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.